1.0/tools/optimize-for-production.md

```diff
@@ -0,0 +1,268 @@
+
+You can pass additional options to vulcanize in the form of flags. For a full list of supported flags, see [the official Vulcanize documentation](https://github.com/Polymer/vulcanize#using-vulcanize-programmatically).
+
+Here’s the same example from above, but this time we’ll tell Vulcanize to strip comments and inline any scripts or CSS.
+
+    vulcanize -o elements.vulcanized.html elements.html --strip-comments --inline-scripts --inline-css
+
+### Use vulcanize with gulp
+
+Although Vulcanize does a great job of flattening imports, you may have an existing build system setup that needs to uglify/minify your code or run your CSS through a preprocessor. This section shows you how to add vulcanize to gulp using  [gulp-vulcanize](https://github.com/sindresorhus/gulp-vulcanize).
+
+*Using Grunt?* You can use the [grunt-vulcanize]() task. It supports a similar set of options [maybe link to README?]
+{: .alert .alert-info }
+
+To add vulcanize to our build process:
+
+Install `gulp-vulcanize`
+
+    npm install --save-dev gulp gulp-vulcanize
+
+Require the vulcanize module in your gulpfile and add a task to run it.
+
+    var gulp = require('gulp');
+    var vulcanize = require('gulp-vulcanize');
+
+    gulp.task('vulcanize', function() {
+      return gulp.src(app/elements/elements.html')
+        .pipe(vulcanize())
+        .pipe(gulp.dest('dist/elements'));
+    });
+
+    gulp.task('default', ['vulcanize']);
+
+This sample assumes your project has a single `elements.html` file that imports your other web component dependencies.
+{: .alert .alert-info }
+
+You should now be able to run `gulp` and vulcanize your dependencies.
+
+To configure the task with the same `stripComments`, `inlineScripts`, and `inlineCss` options from above, pass them to the vulcanize task in a configuration object:
+
+    gulp.task('vulcanize', function() {
+      return gulp.src(app/elements/elements.html')
+        .pipe(vulcanize({
+          stripComments: true,
+          inlineScripts: true,
+          inlineCss: true
+        }))
+        .pipe(gulp.dest('dist/elements'));
+    });
+
+Depending on the structure of your app, it may make sense to break it into a few small vulcanized bundles, instead of inlining everything into one file. To prevent specific imports from being inlined in your bundle, use the `excludes` option and pass it an array of file paths or regexes.
+
+    gulp.task('vulcanize', function() {
+      return gulp.src(app/elements/elements.html')
+        .pipe(vulcanize({
+          excludes: ['elements/x-foo.html']
+        }))
+        .pipe(gulp.dest('dist/elements'));
+    });
+
+This prevents the resources from being _inlined_ in the bundle, however, it leaves link tags in the bundle which attempt to import the resources. If you want to completely remove resources from a bundle, including their link tags, include an additional `stripExcludes` array with the same file paths/regexes.
```

Comment:
For the documented use case (making bundles), I'm not sure the example gives enough detail. 

If you have bundle1.html => 'a.html b.html c.html' and bundle2.html => 'd.html e.html f.html'
where bundle1.html includes the basic stuff loaded on every page and bundle2 is always loaded after bundle1, you could do something like:

bundle2.html: `excludes: ['a.html', 'b.html', 'c.html'], stripExcludes: ['a.html', 'b.html', 'c.html']` ...?

That is, it might not be obvious to the user that they need to strip out the imports by their _original_ name, they can't just exclude `bundle1.html`, because the elements being vulcanized link to a, b, and c, and Vulcanize isn't smart enough to extrapolate. I don't think.